File: clientapp/src/DAL/Dtos/UserDto.ts
import {EntityBase} from "../Models/EntityBase";

export interface User extends EntityBase {
    email: string,
    login: string,
    passwordHash: string,
    /** Фамилия **/
    lastName: string,
    /** Имя **/
    firstName: string,
    /** Отчество **/
    middleName: string,
    /** Пол **/
    gender: string
    age: number
}

File: Backend/Backend.Common/Dtos/UserDto.cs
namespace Backend.Common.Dtos
{
    public class UserDto
    {
        public int Id { get; set; }
        
        public string Email { get; set; }

        public string Login { get; set; }

        /// <summary>
        ///     Пароль
        /// </summary>
        public string PasswordHash { get; set; }

        /// <summary>
        ///     Фамилия
        /// </summary>
        public string LastName { get; set; }

        /// <summary>
        ///     Имя
        /// </summary>
        public string FirstName { get; set; }

        /// <summary>
        ///     Отчество
        /// </summary>
        public string MiddleName { get; set; }

        /// <summary>
        ///     Пол
        /// </summary>
        public string Gender { get; set; }

        /// <summary>
        ///     Возраст
        /// </summary>
        public int Age { get; set; }
    }
}